src/Http/WebUtilities/src/MultipartSectionPipeReader.cs

```diff
@@ -180,7 +180,7 @@ public override async ValueTask<ReadResult> ReadAsync(CancellationToken cancella
                 if (!_boundaryFound)
                 {
                     // We must be examining a partial boundary match, keep reading.
-                    _pipeReader.AdvanceTo(_unconsumedData.Start, _unconsumedData.End);
+                    _pipeReader.AdvanceTo(readResult.Buffer.Start, _unconsumedData.End);
```

Comment:
This was originally _bodyBuffer.Start - I changed it to _unconsumedData.Start, which is totally wrong because it advanced the pipe without the user advancing it.

The reason I changed it was that _bodyBuffer.Start was resulting in a never ending loop - that happend because if the read isn't long enough to check if it's a boundary - it wouldn't set the _bodyBuffer (which does make sense)  - since _bodyBuffer was unset, the advanceTo wouldn't actually advance, and create a never ending loop.

I think that readResult.Buffer.Start would avoid this issue, and reflects the actual meaning (that's why I changed all advances to use it instead of _bodyBuffer.Start - Just informing you so you find it easier to understand the changes I made.